### PR TITLE
Extend a queue's lifetime until async memory operations are complete

### PR DESCRIPTION
**Pull request by fwyzard**

### CODE REVIEW

**Comment by fwyzard** [comment]
This is mostly to test if extending the queue's lifetime would be enough to guarantee that the memory is properly freed.

File: test/unit/mem/buf/src/BufTest.cpp [hunk 1]
```diff
@@ -89,23 +89,19 @@ static auto testAsyncBufferMutable(alpaka::Vec<alpaka::Dim<TAcc>, alpaka::Idx<TA
...
+    // asynchronous operations can be submitted to the queue immediately
+    alpaka::test::testViewMutable<TAcc>(queue, buf);
 
+    // synchronous operations must wait for the memory to be available
     alpaka::wait(queue);
+    auto const offset = alpaka::Vec<Dim, Idx>::zeros();
+    alpaka::test::testViewImmutable<Elem>(buf, dev, extent, offset);
+
+    // the buffer will queue the deallocation of the memory when it goes out of scope,
+    // and extend the lifetime of the queue untul all memory operations have completed.
```
Comment (bernhardmgruber) [inline test/unit/mem/buf/src/BufTest.cpp]:
```suggestion
    // and extend the lifetime of the queue until all memory operations have completed.
```

File: test/unit/mem/buf/src/BufTest.cpp [hunk 2]
```diff
@@ -211,20 +207,16 @@ static auto testAsyncBufferImmutable(alpaka::Vec<alpaka::Dim<TAcc>, alpaka::Idx<
...
+    // memory is allocated when the queue reaches this point
+    auto const buf = alpaka::allocAsyncBuf<Elem, Idx>(queue, extent);
 
+    // synchronous operations must wait for the memory to be available
     alpaka::wait(queue);
+    auto const offset = alpaka::Vec<Dim, Idx>::zeros();
+    alpaka::test::testViewImmutable<Elem>(buf, dev, extent, offset);
+
+    // the buffer will queue the deallocation of the memory when it goes out of scope,
+    // and extend the lifetime of the queue untul all memory operations have completed.
```
Comment (bernhardmgruber) [inline test/unit/mem/buf/src/BufTest.cpp]:
```suggestion
    // and extend the lifetime of the queue until all memory operations have completed.
```

**Review by bernhardmgruber — DISMISSED** [review]

**Review by j-stephan — CHANGES_REQUESTED** [review]
Thanks. Please address Bernhard's comment and we're good to merge.

**Comment by fwyzard** [comment]
I've tested that the change does not have any measurable impact on the performance of our standalone application.

**Review by bernhardmgruber — APPROVED** [review]

**Review by bernhardmgruber — APPROVED** [review]

**Comment by bernhardmgruber** [comment]
@j-stephan please approve and merge!

**Review by j-stephan — APPROVED** [review]

**Pull request merged**